esinstall/src/index.ts

```diff
@@ -341,17 +335,218 @@ export async function install(
   }
   if (Object.keys(installEntrypoints).length === 0 && Object.keys(assetEntrypoints).length === 0) {
     throw new Error(`No ESM dependencies found!
-${colors.dim(
-  `  At least one dependency must have an ESM "module" entrypoint. You can find modern, web-ready packages at ${colors.underline(
-    'https://www.pika.dev',
-  )}`,
-)}`);
+  ${colors.dim(
+    `  At least one dependency must have an ESM "module" entrypoint. You can find modern, web-ready packages at ${colors.underline(
+      'https://www.pika.dev',
+    )}`,
+  )}`);
   }
 
   await initESModuleLexer;
+
+  const bundler = useEsbuild ? bundleWithEsBuild : bundleWithRollup;
+
+  let {importMap: jsFilesImportMap, stats} = await bundler({
+    installEntrypoints,
+    installTargets,
+    ...options,
+  });
+
+  const assetsImportMap: ImportMap = {
+    imports: Object.assign(
+      {},
+      ...Object.keys(assetEntrypoints).map((assetName) => {
+        return {
+          [assetName]: `./${sanitizePackageName(assetName)}`,
+        };
+      }),
+    ),
+  };
+
+  const aliasesImportMap = getAliasesImportMap({allInstallSpecifiers, installAlias});
+
+  // merge all importmaps
+  const importMap: ImportMap = [assetsImportMap, aliasesImportMap, jsFilesImportMap].reduce(
+    merge,
+    {},
+  ) as ImportMap;
+
+  mkdirp.sync(destLoc);
+  await writeLockfile(path.join(destLoc, 'import-map.json'), importMap);
+  for (const [assetName, assetLoc] of Object.entries(assetEntrypoints)) {
+    const assetDest = `${destLoc}/${sanitizePackageName(assetName)}`;
+    mkdirp.sync(path.dirname(assetDest));
+    fs.copyFileSync(assetLoc, assetDest);
+  }
+
+  return {
+    importMap,
+    stats,
+  };
+}
+
+function getAliasesImportMap({installAlias, allInstallSpecifiers}) {
+  const importMap: ImportMap = {imports: {}};
+  for (const installSpecifier of allInstallSpecifiers) {
+    Object.entries(installAlias)
+      .filter(([, value]) => value === installSpecifier)
+      .forEach(([key]) => {
+        const targetName = getWebDependencyName(installSpecifier);
+        // TODO the alias could not be at this path, the target path should be generated based on existing importMap
+        importMap.imports[key] = `./${targetName}.js`;
+      });
+  }
+  return importMap;
+}
+
+type BundlerOptions = PublicInstallOptions & {
+  installEntrypoints: Record<string, string>;
+  installTargets: InstallTarget[];
+};
+
+function makeTsConfig({alias}) {
```

Comment:
This is very cool. Providing aliases just for a TypeScript build then discarding them. 😎

Reply:
Another better approach would be to use the plugin system but it is still unavailable